libraries/AP_Compass/AP_Compass_LIS3MDL.h

```diff
@@ -25,6 +25,7 @@
 #ifndef HAL_COMPASS_LIS3MDL_I2C_ADDR
 // this can also be on 0x1e
 # define HAL_COMPASS_LIS3MDL_I2C_ADDR 0x1c
+# define HAL_COMPASS_LIS3MDL_I2C_ADDR2 0x1e
```

Comment:
This should have a separate ifndef, otherwise it just takes HAL_COMPASS_LIS3MDL_I2C_ADDR being defined for HAL_COMPASS_LIS3MDL_I2C_ADDR2 to be undefined and break the build.